### PR TITLE
Add build.yml workflow

### PR DESCRIPTION
**Pull request by ellet0**
I noticed when we make typos or build errors the code, we can still send the PR and there is no CI that check the code build, so I added this simple GitHub workflow called `build.yml` to help us with this

It will work on all PRs and only when push to the main branch.

### CODE REVIEW

**Comment by ellet0** [comment]
See? we can get the benefit of this PR already, we got caught a test failure in this PR

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,31 @@
+name: Build
+
+on:
+  push:
```
Comment (bitspittle) [inline .github/workflows/build.yml:4]: Let's just do this on pull_requests for now.

I'd like to review it in that context first and then we can extend to push actions later.

**Review by bitspittle — CHANGES_REQUESTED** [review]
Looks good, thanks!

**Comment by ellet0** [comment]
> test-all.yml

I think both your name and mine doesn't fit but it doesn't much matter because we don't have much workflows, just two until now, I will rename it

File: .github/workflows/build.yml (hunk 1 above)
Comment (ellet0) [reply .github/workflows/build.yml]: > Let's just do this on pull_requests for now.
> 
> I'd like to review it in that context first and then we can extend to push actions later.

Done

**Comment by ellet0** [comment]
> Should the file here be called "test-all.yml" or something instead of "build.yml"?

Renamed

**Pull request closed without merging**